gtclang/src/gtclang/Frontend/GTClangASTConsumerParseOnly.cpp

```diff
@@ -0,0 +1,137 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/AST/GridType.h"
+#include "dawn/CodeGen/TranslationUnit.h"
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Serialization/SIRSerializer.h"
+#include "dawn/Support/FileSystem.h"
+#include "dawn/Support/Format.h"
+#include "dawn/Support/Logging.h"
+#include "gtclang/Frontend/ClangFormat.h"
+#include "gtclang/Frontend/Diagnostics.h"
+#include "gtclang/Frontend/GTClangASTAction.h"
+#include "gtclang/Frontend/GTClangASTConsumerParseOnly.h"
+#include "gtclang/Frontend/GTClangASTVisitor.h"
+#include "gtclang/Frontend/GTClangContext.h"
+#include "gtclang/Support/ClangCompat/FileSystem.h"
+#include "gtclang/Support/ClangCompat/VirtualFileSystem.h"
+#include "gtclang/Support/Config.h"
+#include "gtclang/Support/FileUtil.h"
+#include "clang/AST/Decl.h"
+#include "clang/Basic/FileManager.h"
+#include "clang/Basic/SourceLocation.h"
+#include "clang/Basic/SourceManager.h"
+#include "clang/Lex/Lexer.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include "llvm/ADT/IntrusiveRefCntPtr.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/raw_os_ostream.h"
+#include "llvm/Support/raw_ostream.h"
+#include <ctime>
+#include <iostream>
+#include <memory>
+#include <string>
+
+namespace gtclang {
+
+GTClangASTConsumerParseOnly::GTClangASTConsumerParseOnly(GTClangContext* context,
+                                                         const std::string& file,
+                                                         GTClangASTAction* parentAction)
+    : context_(context), file_(file), parentAction_(parentAction) {
+  DAWN_LOG(INFO) << "Creating ASTVisitor ... ";
+  visitor_ = std::make_unique<GTClangASTVisitor>(context);
+}
+
+void GTClangASTConsumerParseOnly::HandleTranslationUnit(clang::ASTContext& ASTContext) {
```

Comment:
Is the code repetition here a temporary solution?
If i understand correctly the whole picture of modularization, there should be an executable (a bash script?) that combines gtc-parse + dawn-opt + dawn-codegen and that should correspond to what currently the gtclang executable does. If that's correct, at the end we would not have GTClangASTConsumer but just this parse only version.

Reply:
All correct. I actually found a way to remove the duplication now, so this won't be an issue!